prefect_kubernetes/jobs.py

```diff
@@ -0,0 +1,295 @@
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    **kube_kwargs: Optional[Dict],
+) -> V1Job:
+    """Task for creating a namespaced Kubernetes job.
+
+    Args:
+        body: A dictionary representing a Kubernetes V1Job specification.
+        namespace: The Kubernetes namespace to create this job in.
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        **kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+
+    Returns:
+        A Kubernetes `V1Job` object.
+
+    Example:
+        Create a job in the default namespace:
+        ```python
+        from prefect import flow
+        from prefect_kubernetes.jobs import create_namespaced_job
+
+        @flow
+        def kubernetes_orchestrator():
+            v1_job_metadata = create_namespaced_job(
+                body={"metadata": {"name": "test-job"}},
+                kubernetes_credentials=KubernetesCredentials.load("k8s-creds"),
+            )
+        ```
+    """
+    api_client = kubernetes_credentials.get_batch_client()
+
+    return await run_sync_in_worker_thread(
+        api_client.create_namespaced_job, namespace=namespace, body=body, **kube_kwargs
+    )
+
+
+@task
+async def delete_namespaced_job(
+    job_name: str,
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    delete_option_kwargs: Optional[Dict] = None,
+    **kube_kwargs: Optional[Dict[str, Any]],
+) -> V1Status:
+    """Task for deleting a namespaced Kubernetes job.
+
+    Args:
+        job_name: The name of a job to delete.
+        namespace: The Kubernetes namespace to delete this job in.
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        delete_option_kwargs: Optional keyword arguments to pass to
+            the V1DeleteOptions object (e.g. {"propagation_policy": "...",
+            "grace_period_seconds": "..."}.
+        **kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+
+    Returns:
+        A Kubernetes `V1Status` object.
+
+    Example:
+        Delete "my-job" in the default namespace:
+        ```python
+        from prefect import flow
+        from prefect_kubernetes.jobs import delete_namespaced_job
+
+        @flow
+        def kubernetes_orchestrator():
+            v1_job_status = delete_namespaced_job(
+                job_name="my-job",
+                kubernetes_credentials=KubernetesCredentials.load("k8s-creds"),
+            )
+        ```
+    """
+
+    api_client = kubernetes_credentials.get_batch_client()
+
+    if delete_option_kwargs:
+        kube_kwargs.update(body=client.V1DeleteOptions(**delete_option_kwargs))
```

Comment:
```suggestion
        kube_kwargs.update(body=client.V1DeleteOptions(**delete_option_kwargs))
```

I don't think we have client defined here.

Reply:
this is just an import from `kubernetes.client`, but this is outdated anyways